fhevm/operators_arithmetic.go

```diff
@@ -3,11 +3,190 @@ package fhevm
 import (
 	"encoding/hex"
 	"errors"
+	"math/big"
 
 	"github.com/ethereum/go-ethereum/common"
+	"github.com/zama-ai/fhevm-go/sgx"
 	"go.opentelemetry.io/otel/trace"
 )
 
+func sgxAddRun(environment EVMEnvironment, caller common.Address, addr common.Address, input []byte, readOnly bool, runSpan trace.Span) ([]byte, error) {
+	input = input[:minInt(65, len(input))]
+
+	logger := environment.GetLogger()
+
+	lhs, rhs, err := get2VerifiedOperands(environment, input)
+	otelDescribeOperands(runSpan, encryptedOperand(*lhs), encryptedOperand(*rhs))
+	if err != nil {
+		logger.Error("sgxAdd inputs not verified", "err", err, "input", hex.EncodeToString(input))
+		return nil, err
+	}
+	if lhs.fheUintType() != rhs.fheUintType() {
+		msg := "sgxAdd operand type mismatch"
+		logger.Error(msg, "lhs", lhs.fheUintType(), "rhs", rhs.fheUintType())
+		return nil, errors.New(msg)
+	}
+
+	// If we are doing gas estimation, skip execution and insert a random ciphertext as a result.
+	if !environment.IsCommitting() && !environment.IsEthCall() {
+		return importRandomCiphertext(environment, lhs.fheUintType()), nil
+	}
+
+	lb, err := sgx.FromTfheCiphertext(*lhs.ciphertext)
+	if err != nil {
+		logger.Error("sgxAdd failed", "err", err)
+		return nil, err
+	}
+
+	rb, err := sgx.FromTfheCiphertext(*rhs.ciphertext)
+	if err != nil {
+		logger.Error("sgxAdd failed", "err", err)
+		return nil, err
+	}
+
+	l := big.NewInt(0).SetBytes(lb.Plaintext).Uint64()
```

Comment:
Propose to add a comment on why we choose to use big.Int (instead of binary directly:

```suggestion
	// Using math/big here to make code more readable.
	// A more efficient way would be to use binary.BigEndian.UintXX().
	// However, that would require a switch case. We prefer for now to use
	// big.Int as a one-liner that can handle variable-length bytes.
	l := big.NewInt(0).SetBytes(lb.Plaintext).Uint64()
```